internal/internal_task_handlers.go

```diff
@@ -651,14 +660,62 @@ ProcessEvents:
 		}
 	}
 
-	completeRequest = wth.completeWorkflow(eventHandler, task, workflowContext, decisions)
+	completeRequest = workflowContext.CompleteDecisionTask()
 
 	if emitStack {
 		stackTrace = eventHandler.StackTrace()
 	}
 	return completeRequest, stackTrace, nil
 }
 
+func (wth *workflowTaskHandlerImpl) ProcessLocalActivityResult(lar *localActivityResult) (interface{}, error) {
+	workflowContext := lar.task.wc
+	if workflowContext.currentDecisionTask != lar.task.decisionTask {
+		// The possible case is decision task timeout while waiting for local activity to complete, then server would
+		// generate a new new decision task, which could be dispatched to this same worker. In that case, the cached
+		// workflowExecutionContext will be reset because the events mismatch and the currentDecisionTask will be updated
+		// to the new decision task while the reference in the local activity task will still be the old one.
+		wth.logger.Debug("Discard local activity result because decision task that it belongs to is no longer available. " +
+			"Possible cause is local activity took too long and decision task timedout.")
+		return nil, nil
+	}
+
+	workflowContext.Lock()
+	defer workflowContext.release()
+
+	eventDecisions, err := workflowContext.eventHandler.ProcessLocalActivityResult(lar)
+	if err != nil {
+		return nil, err
+	}
+
+	if eventDecisions != nil {
+		workflowContext.newDecisions = append(workflowContext.newDecisions, eventDecisions...)
+	}
+
+	return workflowContext.CompleteDecisionTask(), nil
```

Comment:
I think I'm missing  something, but I cannot find where eventLoop is called when a local activity completes.